pkg/operator/controllers/internetchecker/internetchecker_controller.go

```diff
@@ -62,64 +71,76 @@ func (r *InternetChecker) Reconcile(request ctrl.Request) (ctrl.Result, error) {
 		return reconcile.Result{}, err
 	}
 
-	var condition status.ConditionType
-	switch r.role {
-	case operator.RoleMaster:
-		condition = arov1alpha1.InternetReachableFromMaster
-	case operator.RoleWorker:
-		condition = arov1alpha1.InternetReachableFromWorker
-	}
+	// limit all checks to take no longer than half of the requeueInterval
+	ctx, cancel := context.WithTimeout(context.Background(), requeueInterval/2)
+	defer cancel()
 
-	urlErrors := map[string]string{}
+	checks := make(map[string]chan error)
```

Comment:
A map of channels is overkill and makes the concurrency harder to reason about.  Instantiate a single channel used by all the goroutines.  Have checkWithRetry return the full "url: error" text so that you don't need to recombine the url into the error.